File: src/main/java/org/goodgoodgood/freetubeexservice/domain/music/MusicPlaylist.java
package org.goodgoodgood.freetubeexservice.domain.music;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.goodgoodgood.freetubeexservice.domain.common.BaseTimestampEntity;

import javax.persistence.*;

@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@Entity
public class MusicPlaylist extends BaseTimestampEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "music_id")
    private Music music;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "playlist_id")
    private Playlist playlist;

    @Builder
    public MusicPlaylist(Music music, Playlist playlist) {
        this.music = music;
        this.playlist = playlist;
    }
}

File: src/main/kotlin/org/goodgoodgood/freetubeexservice/domain/music/MusicPlaylist.kt
package org.goodgoodgood.freetubeexservice.domain.music

import org.goodgoodgood.freetubeexservice.domain.common.BaseTimestampEntity
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne

@Entity
class MusicPlaylist(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @ManyToOne
    @JoinColumn(name = "music_id")
    val music: Music,

    @ManyToOne
    @JoinColumn(name = "playlist_id")
    val playlist: Playlist,
): BaseTimestampEntity()